### PR TITLE
TAO-7152 - CSP header configuration UI

### PR DESCRIPTION
**Pull request by martijn-tao**
Added new page under settings called "Security". On this page you will find a form that allows you to set the CSP header policy for the platform.

Also added a new key value persistence for the settings. This was done to avoid the usage of RDF for reading and writing simple platform settings.

### CODE REVIEW

File: views/js/controller/routes.js [hunk 1]
```diff
@@ -53,6 +53,11 @@ define(function(){
                 'index' : 'controller/users/index',
                 'add'   : 'controller/users/add'
             }
-        }
+        },
+        'Security' : {
+            'actions' : {
+                'index' : 'form/cspHeader'
+            }
```
Comment (oatymart) [inline views/js/controller/routes.js]: A more suitable place to put the Javascript file we are going to call would be:
```
'Security' : {
            'actions' : {
                'index' : 'controller/security/cspHeaderForm'
            }
}
```

File: views/templates/security/view.tpl [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        <?= get_data('cspHeaderForm'); ?>
+    </div>
+    <?php endif;?>
+</div>
+
+<?php if (get_data('cspHeaderFormSuccess')): ?>
+<div id="csp-header-success" class="hidden"><?= get_data('cspHeaderFormSuccess') ?></div>
+<?php endif;?>
+
+<script>
```
Comment (oatymart) [inline views/templates/security/view.tpl]: The contents of the `<script>` block should come out of this template and into the controller we defined in the routes (`/tao/views/js/controller/security/cspHeaderForm.js`)

File: actions/form/class.CspHeader.php [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+     */
+    private function setFormData()
+    {
+        $currentSetting = $this->getSettings();
+        $listSettings = [];
+        if ($currentSetting === 'list') {
+            $listSettings = $this->getListSettings();
+        }
+
+        if (!isset($_POST[self::SOURCE_RADIO_NAME]) && $currentSetting) {
```
Comment (alexeyvy) [inline actions/form/class.CspHeader.php]: Can we avoid using global variables here? Maybe $this->data or something?

File: config/default/settingsStorage.conf.php [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?php
```
Comment (alexeyvy) [inline config/default/settingsStorage.conf.php]: The service id is 'SettingStorage' while this file is 'settingsStorage' (one more 's')

File: helpers/form/validators/CspHeaderValidator.php [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+ */
+
+namespace oat\tao\helpers\form\validators;
+
+/**
+ * Validates the given CSP headers
+ *
+ * @author Martijn Swinkels <m.swinkels@taotesting.com>
+ */
+class CspHeaderValidator extends \tao_helpers_form_Validator
```
Comment (alexeyvy) [inline helpers/form/validators/CspHeaderValidator.php]: This class contains much logic in it, therefore I think should be covered with some tests (more precisely, evaluate() as a single public function and an entry point to all the logic is a target for comprehensive testing :) )

File: models/classes/settings/CspHeaderSettingsInterface.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+namespace oat\tao\model\settings;
+
+/**
+ * Interface CspHeaderSettingsInterface
+ *
+ * @author Martijn Swinkels <m.swinkels@taotesting.com>
+ */
+interface CspHeaderSettingsInterface
```
Comment (alexeyvy) [inline models/classes/settings/CspHeaderSettingsInterface.php]: maybe use interface inheritance here? I mean `CspHeaderSettingsInterface` abstraction always extends `SettingsStorageInterface`

File: actions/form/class.CspHeader.php (hunk 3 above)
Comment (martijn-tao) [reply actions/form/class.CspHeader.php]: I believe the posted form data is currently only available through the `$_POST` varaible. I could modify the form to add the post data to a class field however. What are your thoughts @alexeyvy?

File: models/classes/settings/CspHeaderSettingsInterface.php (hunk 6 above)
Comment (martijn-tao) [reply models/classes/settings/CspHeaderSettingsInterface.php]: I'm only using this interface to define the constants used by this specific configuration. The `\oat\tao\model\settings\SettingsStorageInterface` actually defines the implementation of the settingsStorage

**Pull request merged**